Source/DOH/file.c

```diff
@@ -286,21 +286,36 @@ DOH *DohNewFile(DOHString *filename, const char *mode, DOHList *newfiles) {
   char *filen;
 
   filen = Char(filename);
+
   file = fopen(filen, mode);
-  if (!file)
-    return 0;
+  if (!file) {
+    return NULL;
+  }
 
   f = (DohFile *) DohMalloc(sizeof(DohFile));
+  if (!f) {
+    fclose(file);
+    return NULL;
+  }
```

Comment:
`DohMalloc()` prints an error and exits if memory allocation fails so it's not necessary or useful to check like this.